[quant][embedding QAT] Add Embedding op with fused FakeQuant

b-koopman opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68900
* #67945
* __->__ #66985
* #68296
* #68390

Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

b-koopman committed:
[quant][embedding QAT] Add Embedding op with fused FakeQuant

[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"




[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]

jingsh commented on aten/src/ATen/native/Embedding.cpp:
```diff
@@ -9,6 +9,7 @@
 #include <memory>
 #include <sstream>
 #include <vector>
+#include <iostream>
```
remove this, it's included but not used

jingsh commented on torch/nn/qat/modules/embedding_ops.py:
```diff
@@ -69,6 +115,20 @@ def to_float(self):
         embedding_bag.train(self.training)
         return embedding_bag
+
+class FusedFakeQuantEmbedding(Embedding):
+    def __init__(self, num_embeddings, embedding_dim, padding_idx=None,
+                 max_norm=None, norm_type=2.0, scale_grad_by_freq=False,
+                 sparse=False, _weight=None, device=None, dtype=None, qconfig=None) -> None:
+                 super().__init__(num_embeddings, embedding_dim, padding_idx, max_norm,
```
probably good to run auto format on this file, the format does not look good

jingsh reviewed (commented):
There's CI failures, can you look at them, and address some of these lint

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]

b-koopman committed:
Update on "[quant][embedding QAT] Add Embedding op with fused FakeQuant"


Differential Revision: [D32012973](https://our.internmc.facebook.com/intern/diff/D32012973)

[ghstack-poisoned]
